Task 3/Task 3.2/Task 3.2/Task 3.2/DynamicArray.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Task_3._2
+{
+    public class DynamicArray<T> : IEnumerable<T>, ICloneable
+    {
+        private protected T[] _myArray { get; set; }
+
+        public int capacity { get; private set; } = 8;
+
+        public int length { get; private set; } = 0;
+
+        public DynamicArray()
+        {
+            _myArray = new T[capacity];
+        }
+
+        public DynamicArray(int value)
+        {
+            if (value > capacity)
+            {
+                while (capacity < value)
+                {
+                    capacity *= 2;
+                }
+            }
+
+            _myArray = new T[capacity];
+        }
+
+        public DynamicArray(ICollection<T> list)
+        {
+            if (list.Count > capacity)
+            {
+                while (capacity < list.Count)
+                {
+                    capacity *= 2;
+                }
```

Comment:
This can be done without `while` loop (by calculating multiplicator or setting capacity to list length + some buffer)